chore(substrait): update substrait to version 0.8

gforsyth opened the pull request.

cpcloud reviewed (approved):
Should this be a `fix` commit to bump the version?

gforsyth commented:
> Should this be a `fix` commit to bump the version?

I didn't realize that the auto-release was predicated on the commit types, but that makes sense.  Rebased and updated
